CoreAlgorithmPlugins/test/unit/src/au/gov/asd/tac/constellation/plugins/algorithms/clustering/infomap/InfoMapPluginNGTest.java

```diff
@@ -0,0 +1,234 @@
+     */
+    @Test
+    public void testCreateConfig() {
+        System.out.println("createConfig");
+
+        final String[] connectionTypes = {
+            CONNECTION_TYPE_TRANSACTIONS,
+            CONNECTION_TYPE_EDGES,
+            CONNECTION_TYPE_LINKS
+        };
+
+        final String[] dynamicTypes = {
+            DYNAMICS_PARAMETER_UNDIRECTED,
+            DYNAMICS_PARAMETER_DIRECTED,
+            DYNAMICS_PARAMETER_UNDIRECTED_FLOW,
+            DYNAMICS_PARAMETER_INCLOMING_FLOW,
+            DYNAMICS_PARAMETER_DIRECTED_WEIGHT
+        };
+
+        final InfoMapPlugin instance = new InfoMapPlugin();
+        for (final String connection : connectionTypes) {
+            for (final String dynamic : dynamicTypes) {
+                final PluginParameters params = createParametersHelper(connection, dynamic);
+                final Config config = instance.createConfig(params);
+
+                // Assert connection and dynamic was set in config
+                configParameterHelper(connection, dynamic, config);
+            }
+        }
+
+    }
+
+    private void configParameterHelper(final String connectionType, final String dynamicType, final Config config) {
+        // Connection type.
+        final Config.ConnectionType connectionTypeEnum = connectionTypeHelper(connectionType);
+        assertEquals(connectionTypeEnum, config.getConnectionType());
+
+        // Dynamic type
+        // Expected true if given dynamic type matches this dynamic type
+        final boolean expResultUndirected = dynamicType.equals(DYNAMICS_PARAMETER_UNDIRECTED);
+        final boolean expResultDirected = dynamicType.equals(DYNAMICS_PARAMETER_DIRECTED);
+        final boolean expResultUndirectedFlow = dynamicType.equals(DYNAMICS_PARAMETER_UNDIRECTED_FLOW);
+        final boolean expResultIncomingFlow = dynamicType.equals(DYNAMICS_PARAMETER_INCLOMING_FLOW);
+        final boolean expResultDirectectWeigth = dynamicType.equals(DYNAMICS_PARAMETER_DIRECTED_WEIGHT);
+
+        assertEquals(expResultUndirected, config.isUndirected());
+        assertEquals(expResultDirected, config.isDirected());
+        assertEquals(expResultUndirectedFlow, config.isUndirdir());
+        assertEquals(expResultIncomingFlow, config.isOutdirdir());
+        assertEquals(expResultDirectectWeigth, config.isRawdir());
+    }
+
+    private Config.ConnectionType connectionTypeHelper(final String connectionType) {
+        return switch (connectionType) {
+            case CONNECTION_TYPE_TRANSACTIONS ->
+                Config.ConnectionType.TRANSACTIONS;
+            case CONNECTION_TYPE_EDGES ->
+                Config.ConnectionType.EDGES;
+            case CONNECTION_TYPE_LINKS ->
+                Config.ConnectionType.LINKS;
+            default ->
+                Config.ConnectionType.TRANSACTIONS;
+        };
+    }
+
+    private PluginParameters createParametersHelper(final String connectionType, final String dynamicType) {
```

Comment:
this code in this function can probably be simplified by calling instance.createParameters() and then just updating the parameter values with the desired values, rather than recreating them all from scratch